packages/core/src/discovery.js

```diff
@@ -203,6 +204,20 @@ async function* captureSnapshotResources(page, snapshot, options) {
   yield resizePage(snapshot.widths[0]);
   yield page.goto(snapshot.url, { cookies });
 
+  // wait for any specified timeout
+  if (snapshot.discovery.waitForTimeout && snapshot.enableJavaScript) {
```

Comment:
we are checking for js enabled assets discovery browser, check for `page.enableJavascript` that will have condition checked if the assets discovery browser is opened in js enabled or not.